qaframework-bdd-tests/src/test/java/org/openmrs/contrib/qaframework/automation/AllergiesSteps.java

```diff
@@ -65,4 +82,71 @@ public void removeNoKnownAllergy() {
 	public void systemRemovesNoKnownAllergies() {
 		assertTrue(textExists("Unknown"));
 	}
+	
+	@When("a user clicks Add New Allergy button")
+	public void loadAddAllergyPage() {
+		addOrEditAllergyPage = allergyPage.clickOnAddNewAllergy();
```

Comment:
Are you instantiating addOrEditAllergyPage here ?  . what happens when you simply call, `allergyPage.clickOnAddNewAllergy();`

Reply:
@sherrif10 at that point basically by clicking the Add New Allergy button, the the system is asked to load the addOrEditAllergyPage. So if it is not instantiated here then line 99 will fail. It will throw a null pointer exception, because the page  addOrEditAllergyPage will not be available. What are your thoughts @sherrif10 ?

Reply:
Have no objection. Thanks.

Reply:
Thanks @sherrif10 . Are there any more reviews? :-)